lib/magma/attribute.rb

```diff
@@ -3,12 +3,16 @@ class Attribute
     DISPLAY_ONLY = [:child, :collection]
     attr_reader :name, :type, :desc, :loader, :match, :format_hint, :unique, :index, :restricted
 
-
     class << self
       def options
         [:type, :desc, :display_name, :hide, :readonly, :unique, :index, :match,
 :format_hint, :loader, :link_model, :restricted ]
       end
+
+      def set_attribute(name, model, options, caller)
+        klass = "Magma::#{caller.to_s.capitalize}_attribute".camelcase.constantize
```

Comment:
Could probably move this stuff to the #define_method call below and avoid the caller argument.

Reply:
👍🏼